libs/core/src/main/java/org/opensearch/core/common/io/stream/BaseWriteable.java

```diff
@@ -54,6 +55,13 @@ class WriteableRegistry {
             READER_REGISTRY.put(ordinal, reader);
         }
 
+        public static void registerWriterCustomClass(final Class<?> classInstance, final Class<?> classGeneric) {
+            if (WRITER_CUSTOM_CLASS_MAP.containsKey(classInstance)) {
```

Comment:
Since this is `ConcurrentMap`, put & check could be done in one shot:
```suggestion
            if (WRITER_CUSTOM_CLASS_MAP.putIfAbsent(classInstance, classGeneric) != null) {
```